extension-workspace-references.md

```diff
@@ -0,0 +1,182 @@
+}
+```
+
+_Response_:
+* result: `ReferenceInformation[]` defined as follows:
+```typescript
+/**
+ * Represents information about a reference to programming constructs like
+ * variables, classes, interfaces etc.
+ */
+interface ReferenceInformation {
+    /**
+     * The location in the workspace where the `symbol` has been referenced.
+     */
+    reference: Location;
+
+    /**
+     * Metadata information describing the symbol being referenced.
+     */
+    symbol: SymbolDescriptor;
+}
+```
+* error: code and message set in case an exception happens during the workspace references request.
+
+Where `SymbolDescriptor` is defined as follows:
+
+```typescript
+/**
+ * Represents information about a programming construct like a variable, class,
+ * interface etc that has a reference to it. Effectively, it contains data similar
+ * to SymbolInformation except all fields are optional.
+ *
+ * SymbolDescriptor usually uniquely identifies a symbol, but it is
+ * not guaranteed to do so.
+ */
+interface SymbolDescriptor {
+    /**
+     * The name of this symbol (same as `SymbolInformation.name`).
+     */
+    name?: string;
+
+    /**
+     * The kind of this symbol (same as `SymbolInformation.kind`).
+     */
+    kind?: number;
+
+    /**
+     * The URI of this symbol (same as `SymbolInformation.location.uri`).
+     */
+    uri?: string;
+
+    /**
+     * The container of this symbol. It is up to the language server to define
+     * what exact data this object contains.
+     */
+    container?: Object;
+
+    /**
+     * Whether or not the symbol is defined inside of "vendored" code. In Go, for
+     * example, this means that an external dependency was copied to a subdirectory
+     * named `vendor`. The exact definition of vendor depends on the language,
+     * but it is generally understood to mean "code that was copied from its
+     * original source and now lives in our project directly".
+     */
+    vendor?: boolean;
+
+    /**
+     * Information about the package/library that this symbol is defined in.
+     */
+    package: PackageDescriptor;
+
+    /**
+     * Attributes describing the symbol that is being referenced. It is up to the
+     * language server to define what exact data this object contains.
+     */
+    attributes: Object;
+}
+```
+
+Where `PackageDescriptor` is defined as follows:
+
+```
+/**
+ * Represents information about a programming code unit like a package,
+ * library, crate, module, etc. It uniquely identifies a package at a specific
+ * version within a given registry.
+ */
+interface PackageDescriptor {
+    /**
+     * An ID representing the package of code. For example, in JS this would be
+     * the NPM package name. In Go, the full import path. etc.
+     */
+    id: string;
+
+    /**
+     * The version of the package in the registry.
+     */
+    version?: VersionDescriptor;
+
+    /**
+     * The registry for this package. Examples:
+     *
+     *  - JS: "npm"
```

Comment:
Is the registry really npm, or is npm just a type of registry? The registry type is "npm" and the registry is "npmjs.com", etc. For Go, the registry type would be "go get" and the registry would be empty.

Analogy: This is like if there was a "repo" field and the values were "git", "hg", etc. You'd expect the values to be the clone URL.